### ISSUE TITLE
Update spinal levels based on Frostell et al. paper

### ISSUE THREAD

**Issue by mguaypaq** [spinalcordtoolbox/spinalcordtoolbox#4215]
This is just a tracking issue for the PAM50 levels update, which is happening in the PAM50 repository, but which we want to synchronize with the upcoming SCT v6.1 release. Links:
- [x] The issue: [PAM50#16](https://github.com/spinalcordtoolbox/PAM50/issues/16)
- [x] The pull request: [PAM50#18](https://github.com/spinalcordtoolbox/PAM50/pull/18)
- [x] https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4246
- [x] https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4247
- [ ] https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4248
- [x] Update the PAM50/template/info_label.txt to add the spinal levels
  - [x] Figure out what's going on with the entry "14, point-wise spinal level labels, PAM50_label_spinal_levels.nii.gz"
- [x] Create new PAM50 release
- [x] Update `sct_download_data` to point at the new PAM50 release:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2037cdc135ee891d634b17469a9f0f45d8734624/spinalcordtoolbox/download.py#L47-L53

**Comment by jcohenadad** [comment]
@joshuacwnewton would you mind helping with:
- Create new PAM50 release
- Update sct_download_data to point at the new PAM50 release:
Thanks!

**Comment by joshuacwnewton** [comment]
Of course! :)

**Comment by joshuacwnewton** [comment]
There is one last task:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4248

However, this does not need to be done before SCT v6.1. since it does not involve any changes to SCT's codebase. So, we can treat it like a separate issue.

Since all of the main user-facing changes have been completed, I'm going to close this issue (for the sake of issue tracking for 6.1).
